ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -975,6 +977,138 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_chain_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+
+    def _on_span_started(span: Span):
+        inputs = get_argument_value(args, kwargs, 0, "input")
+        if not integration.is_pc_sampled_span(span):
+            return
+        if not isinstance(inputs, list):
+            inputs = [inputs]
+        for idx, inp in enumerate(inputs):
+            if not isinstance(inp, dict):
+                span.set_tag_str("langchain.request.inputs.%d" % idx, integration.trunc(str(inp)))
+                continue
+            for k, v in inp.items():
+                span.set_tag_str("langchain.request.inputs.%d.%s" % (idx, k), integration.trunc(str(v)))
+
+    def _on_span_finished(span: Span, streamed_chunks):
+        if span.error or not integration.is_pc_sampled_span(span):
+            return
+        if (
+            streamed_chunks
+            and langchain_core
+            and isinstance(instance.steps[-1], langchain_core.output_parsers.JsonOutputParser)
+        ):
+            # it's possible that the chain has a json output parser
+            # this will have already concatenated the chunks into a json object
+
+            # it's also possible the json output parser isn't the last step,
+            # but one of the last steps, in which case we won't act on it here
+            content = json.dumps(streamed_chunks[-1])
+        else:
+            # best effort to join chunks together
+            content = "".join([str(chunk) for chunk in streamed_chunks])
+        span.set_tag_str("langchain.response.content", integration.trunc(content))
```

Comment:
I think we can rename the tag name to `langchain.response.outputs` since this is the output of the chain, and not necessarily a llm/chat model, right?

Reply:
yeah that makes sense to me!